docs/develop/Architecture.md

```diff
@@ -0,0 +1,77 @@
+# Architecture
+
+There are five repositories in the ElectionGuard SDK.
```

Comment:
It would be super helpful to specify up front that there are two reference implementations.  So maybe like:

> There are five repositories in the ElectionGuard SDK: a human readable reference implementation, a sample user interface, an API, and two reference implementations.

Reply:
Agreed, either up front or described with the link such as: 

- [ElectionGuard Specification][election-guard-specification] - a human readable reference implementation
- [ElectionGuard Python][election-guard-python-source] - a Python reference implementation
- [ElectionGuard C++][election-guard-cpp-source-code] - a C++/.NET reference implementation
- [ElectionGuard Python API][election-guard-web-api-source] - a sample API used by the sample user interface
- [ElectionGuard React UI][election-guard-ui-source] - a sample user interface

I think it would be nice to say that the C++ is also a .NET reference because many .NET programmers do not know that you can use C++ inside of it.

Reply:
@lprichar There are 4 reference implementations. Two paths.

Reply:
> C++ is also a .NET reference

+1, I had no idea.  I recall someone mentioning .NET verbally, but I've seen no reference to it in the docs so far.  So mentioning it here would be nice, but unless I'm missing it, a lot more details somewhere in the docs would be extremely helpful too.  Maybe add an issue to track that work rather than make this PR even bigger @keithrfung?

Reply:
> There are 4 reference implementations. Two paths.

@keithrfung I had no idea.  Please do expand on that in the docs.  I have no idea what you mean by "two paths", so please expand on that.  And please expand on the 4 reference implementations, and how they relate to paths, and how they relate to repositories.

Reply:
I'm going to try to reword this a bit better and find a middle ground.